tests/test_lineshapes.py

```diff
@@ -12,8 +12,7 @@
 
 @pytest.mark.parametrize(
     "value, expected_result",
-    [(1, 1.0), (-1, -1.0), (0, tiny), (-0, -tiny), (np.array([1]), 1.0)],
-)
+    [(1, 1.0), (-1, -1.0), (0, tiny), (-0, -tiny)])
```

Comment:
this doesn't fix a deprecation warning in the `not_zero` method, it merely removes the test that triggers this warning. So unless now the test is suddenly not necessary anymore because we can never get into this situation, it doesn't really seem to fix the underlying problem - correct? 

I've looked a bit at the release notes of the recent SciPy and NumPy versions and I can probably spend some time to get some of this sorted out. But as I said in previous PRs as well, just removing tests because they don't pass anymore or produce warnings doesn't look like the best approach.

Reply:
I don't view this as removing a test because it fails but removing a test that tests something we don't do.  That is, none of the 3 uses of `not_zero()` pass in an array.   We're not checking the data type of the value passed to `not_zero()`  within the function: a string, dict, tuple, list, would also fail.    It turns out that, weirdly, arrays with length=1 used to work.  

There are probably 25 or more places in `lineshapes.py` alone where `non_zero()` could be used but is currently using `val = max(tiny, val)` instead:  None of those would work with anything but floats (or ints, I suppose) either.

Reply:
fair enough, I see that the `dho` lineshape code is now indeed such that `non_zero` is not used on arrays anymore. I *think* initially it (or some variant thereof) was used and that is why the test was there. Anyway, if the conclusion is that the test is not needed anymore, that's okay with me - at the very least then the commit message is rather misleading ;)

If you consider this PR finalized I am happy to take another look at if if you think that's helpful. However, if you're not interested in opinions and/or suggestions and just want to keep merging PRs that's also fine, but then I'll not waste any time on it. Just let me know.

Reply:
OK, thanks, yeah sometimes single-line commit comments are not enough, and good commit comments are hard enough anyway...

Yes, I consider this PR ready to merge.